### DISCUSSION TITLE
FileName Obfuscate

### DISCUSSION THREAD

**Discussion by jgh0721** [winfsp/winfsp#541] [General]
I'm creating a virtual drive using the WinFSP driver and obfuscating the filename to show it. 

This means that to the user, the path Z:\a\b\c.txt is mapped to a random string C:\Sandbox\wqrfsd\ewfdwesd\rewrwerwe when it is actually stored physically. 

The implementation is based on an example called ntptfs, and the CreateFile callback returns success (any internal NtCreateFile or NtOpenFile etc. will be successful), but ultimately Windows returns a 123 (ERROR_INVALID_NAME) error.  

Are there any examples, etc. of how to obfuscate filenames in WinFSP?

**Comment by billziss-gh** [comment]
Your obfuscation scheme must produce valid file names in Windows. Read about valid file names in Windows here: https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file

**Comment by xgjhlws** [comment]
I'm having the same problem now. Real File is `.PlaceHolder.txt`，fake file name is `FakePlaceHolder.txt`. They are all at root dir of virtual disk.

This is simple Test code，GetFileAttributes and CreateFile return error code 123.
![image](https://github.com/user-attachments/assets/64d3e58f-f955-4fee-942a-b63205666f16)

My implementation is also based on example ntptfs, and all related function return 0 (STATUS_SUCCESS).
![image](https://github.com/user-attachments/assets/e7f99bef-f360-48e3-b53f-7efcbc21ba28)

But call CreateFile in other programs return error code 123 (ERROR_INVALID_NAME).

Can you provide some help ? @billziss-gh Thanks.
